litestar/typing.py

```diff
@@ -81,10 +92,11 @@ def _parse_metadata(value: Any, is_sequence_container: bool, extra: dict[str, An
         **cast("dict[str, Any]", extra or getattr(value, "extra", None) or {}),
         **(getattr(value, "json_schema_extra", None) or {}),
     }
+    example_list: list[Any] | None
     if example := extra.pop("example", None):
-        example_list = [Example(value=example)]
+        example_list = [example]
     elif examples := getattr(value, "examples", None):
-        example_list = [Example(value=example) for example in cast("list[str]", examples)]
+        example_list = examples
```

Comment:
Looking at where _we_ use it, changing the type of `KwargDefinition.examples` likely isn't that big a deal, but it is publicly documented as it is, and I don't know that changing its type to `Any` is very good.

I think this also means that `get_formatted_examples()` could receive `Sequence[Any]` now, is that right?

https://github.com/litestar-org/litestar/blob/024dccff1e6bf433d260c7fd2add7694eb05f63f/litestar/_openapi/schema_generation/utils.py#L103

Reply:
The method here picks up stuff from the field declaration, and puts into that dict what is relevant / needed to create... at least `Schema` entry <-- I think my understanding/assumption was that it's only used for `Schema` creation, but if it's also used for something else(?) then there may be something wrong here.

But that's different than the flow that you described.

I'm not changing `KwargDefinition.examples` type and must not change it. It will remain as `list[Example]`, as that's the syntax/format how _OpenAPI_ examples are declared currently and that's fine. But JSON schema `examples` need that `list[Any]` instead (as they're just arrays of values).

I guess the question here is exactly what this `_parse_metadata()` is used for - where is that return value used... I'll check again.

> I find this corner of the library really hard to follow

Yes, I would even describe it as _super_ complex and kludgy 😵‍💫

Reply:
Yeah, I believe you're righting. I reverted that in a fixup commit.

Funnily, changing it one way or around doesn't break anything. For sure there's _some_ impact (in the OpenAPI output), but I guess it's not unit test covered.

Reply:
> but if it's also used for something else(?)

Its used on any occasion that we parse an annotation that has _any object_ that defines those openapi/schema constraint attributes in its metadata, e.g., `query_value: Annotated[int, Parameter(example=...)]` on a handler, and also `field_on_a_model: Annotated[int, Meta(example=...)]` on some domain model. Both of these cases get end up with the example value getting shoehorned into `Example`, which - as your recent work has brought to light - is not a correct abstraction for the case where the example should exist on a json schema document.

However imperfect it is that we are using `Example` to carry around that json schema example value internally here between model parsing and schema generation - I guess "it is what it is" for now.

> Yes, I would even describe it as super complex and kludgy 😵‍💫

`FieldDefinition` used to be called `ParsedType` - and its only purpose was to represent a type annotation. It would handle extraction of meta values out of `Annotated`, but not make any assumptions about how that metadata should be used - just a container for type information.

I don't think we need to do all of this metadata parsing/merging/transforming here because its got nothing to do with parsing type annotations and everything to do with schema generation, so should probably exist in `litestar._openapi`. Problem there is that at the same time `ParsedType` turned into `FieldDefinition`, `FieldDefinition.kwarg_definition` was added to the public api, which is the result of all of this metadata wrangling. So its probably going to have to be a 3.0 thing to unwind.